RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -951,10 +993,81 @@ void PFAlgo::processBlock( const reco::PFBlockRef& blockref,
+    if (hasDeadHcal) {
+       goodTrackDeadHcal = ( trackRef->ptError() < goodTrackDeadHcal_ptErrRel_ * trackRef->pt() &&
+                             trackRef->normalizedChi2() < goodTrackDeadHcal_chi2n_ &&
+                             trackRef->hitPattern().trackerLayersWithMeasurement() >= goodTrackDeadHcal_layers_ &&
+                             trackRef->validFraction() > goodTrackDeadHcal_validFr_ &&
+                             std::abs(trackRef->dxy(primaryVertex_.position())) < goodTrackDeadHcal_dxy_ );
+       // now we add an extra block for tracks at high |eta|
+       if (!goodTrackDeadHcal &&
+            std::abs(trackRef->eta()) > goodPixelTrackDeadHcal_minEta_ && // high eta
+            trackRef->hitPattern().pixelLayersWithMeasurement() >= 3   && // pixel track
+            trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::TRACK_HITS) == 0 &&
+            trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::MISSING_INNER_HITS) == 0 &&
+            trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::MISSING_OUTER_HITS) <= (
+                trackRef->hitPattern().pixelLayersWithMeasurement() > 3 ?
+                goodPixelTrackDeadHcal_maxLost4Hit_ : goodPixelTrackDeadHcal_maxLost3Hit_  ) &&
+            trackRef->normalizedChi2() < goodPixelTrackDeadHcal_chi2n_ && // tighter cut
+            std::abs(trackRef->dxy(primaryVertex_.position())) < goodPixelTrackDeadHcal_dxy_  &&
+            std::abs(trackRef->dz(primaryVertex_.position()))  < goodPixelTrackDeadHcal_dz_  &&
+            trackRef->ptError() < goodPixelTrackDeadHcal_ptErrRel_*trackRef->pt() && // sanity
+            trackRef->pt() < goodPixelTrackDeadHcal_maxPt_) {                        // sanity
+            goodTrackDeadHcal = true;
+            // FIXME: may decide to do something to the track pT
+       }
+       //if (!goodTrackDeadHcal && trackRef->hitPattern().trackerLayersWithMeasurement() == 4 && trackRef->validFraction() == 1
+       if (debug_) cout << " track pt " << trackRef->pt() << " +- " << trackRef->ptError()
+          << " layers valid " << trackRef->hitPattern().trackerLayersWithMeasurement()
+                << ", lost " << trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::TRACK_HITS)
+                << ", lost outer " << trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::MISSING_OUTER_HITS)
+                << ", lost inner " << trackRef->hitPattern().trackerLayersWithoutMeasurement(HitPattern::MISSING_INNER_HITS)
+                << "(valid fraction " << trackRef->validFraction() << ")"
+          << " chi2/ndf " << trackRef->normalizedChi2()
+          << " |dxy| " << std::abs(trackRef->dxy(primaryVertex_.position())) << " +- " << trackRef->dxyError()
+          << " |dz| " << std::abs(trackRef->dz(primaryVertex_.position())) << " +- " << trackRef->dzError()
+          << (goodTrackDeadHcal ? " passes " : " fails ") << "quality cuts" << std::endl;
+    }
```

Comment:
shouldn't there be an 
```else //no dead hcal 
goodTrackDeadHcal = true; 
```

otherwise, it's unclear below if the check on hasDeadHcal is done appropriately ?

Reply:
I'm not sure I understand the question, but goodTrackDeadHcal is a subset of hasDeadHcal